### PR TITLE
Fixes chameleon PDAs not being able to morph into assistant PDAs.

### LINKED ISSUE #22900: Selecting "Flesh" on a chameleon mask displays HONK! on the tooltip icon
easy sprite fix, get your antag tokens here

### LINKED ISSUE #23044: Cham PDA can't select base PDA type
This means you cannot get the default assistant pda sprite. You can select "pda", but that appears to be iconstate "NOTHING" in pda.dmi, which is the ai's pda. but the ai's pda shouldn't even be selectable, as it's blacklisted.

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
fix: Chameleon PDAs can now morph into assistant PDAs.
fix: A few iconless items have been blacklisted from chameleon clothing.
/:cl:

Fixes #23044
Fixes #22900

The pAI PDA and later, the generic heads PDA were hijacking the name "PDA", hiding the assistant PDA.

I also reworked the list to be a typecache, and since for some reason typecaches automatically grabbed typesof or subtypesof the types you inserted, i added an option not to.
Also added a few iconless items to the blacklists.

### CODE REVIEW

File: code/modules/clothing/chameleon.dm [hunk 1]
```diff
@@ -69,6 +69,7 @@
 /datum/action/item_action/chameleon/change
 	name = "Chameleon Change"
 	var/list/chameleon_blacklist = list()
+	var/list/chameleon_blacklist_types = list()
```
Comment (lzimann) [inline code/modules/clothing/chameleon.dm]: the var names are really confusing, plus if they are checking types make it a typecache!

File: code/modules/clothing/chameleon.dm [hunk 2]
```diff
@@ -69,18 +69,28 @@
...
 	var/chameleon_type = null
 	var/chameleon_name = "Item"
 
 	var/emp_timer
 
+/datum/action/item_action/chameleon/change/New()
+	..()
+	chameleon_blacklist = typecacheof(chameleon_blacklist)
+	chameleon_blacklist_with_subtypes = typecacheof(chameleon_blacklist_with_subtypes)
+	chameleon_list = typecacheof(chameleon_list)
```
Comment (lzimann) [inline code/modules/clothing/chameleon.dm]: This won't work for typecache since it won't ever check for types.

Comment (lzimann) [inline code/modules/clothing/chameleon.dm]: Also is it really needed having two vars for blacklisting? It's hard to think a situation where it's needed.

Comment (XDTM) [inline code/modules/clothing/chameleon.dm]: How should i go about it?

Comment (XDTM) [inline code/modules/clothing/chameleon.dm]: About the two types, you're right, they're not necessary, now that i think about it. I'll remove them.

Comment (lzimann) [inline code/modules/clothing/chameleon.dm]: You can keep it as a regular list like it was before.

**Comment by XDTM** [comment]
@AnturK Ready for review

**Pull request merged**